x-pack/test/observability_ai_assistant_functional/common/config.ts

```diff
@@ -5,29 +5,108 @@
-import {
-  ObservabilityAIAssistantFtrConfig,
-  createObservabilityAIAssistantAPIConfig,
-} from '../../observability_ai_assistant_api_integration/common/config';
-import { getScopedApiClient } from '../../observability_ai_assistant_api_integration/common/observability_ai_assistant_api_client';
+import path from 'path';
+import { secondaryEditor, editor, viewer } from './users/users';
+import { getScopedApiClient } from '../../api_integration/deployment_agnostic/apis/observability/ai_assistant/utils/observability_ai_assistant_api_client';
 import { InheritedFtrProviderContext, InheritedServices } from '../ftr_provider_context';
 import { ObservabilityAIAssistantUIProvider } from './ui';
+import { getApmSynthtraceEsClient } from './create_synthtrace_client';
+
+export interface ObservabilityAIAssistantFtrConfig {
+  name: ObservabilityAIAssistantFtrConfigName;
+  license: 'basic' | 'trial';
+  kibanaConfig?: Record<string, any>;
+}
+
+export const observabilityAIAssistantDebugLogger = {
+  name: 'plugins.observabilityAIAssistant',
+  level: 'debug',
+  appenders: ['console'],
+};
+
+export const observabilityAIAssistantFtrConfigs = {
+  basic: {
+    license: 'basic' as const,
+    kibanaConfig: {
+      'logging.loggers': [observabilityAIAssistantDebugLogger],
+    },
+  },
+  enterprise: {
+    license: 'trial' as const,
+    kibanaConfig: {
+      'logging.loggers': [observabilityAIAssistantDebugLogger],
+      'plugin-path': path.resolve(
+        __dirname,
+        '../../../../src/platform/test/analytics/plugins/analytics_ftr_helpers'
+      ),
+    },
+  },
+};
+
+export type ObservabilityAIAssistantFtrConfigName = keyof typeof observabilityAIAssistantFtrConfigs;
+
+export type ObservabilityAIAssistantServices = Awaited<ReturnType<CreateTestConfig>>['services'];
 
 export type CreateTestConfig = ReturnType<typeof createTestConfig>;
 export type TestConfig = Awaited<ReturnType<typeof getTestConfig>>;
 
+export function createObservabilityAIAssistantAPIConfig({
+  config,
+  license,
+  name,
+  kibanaConfig,
+}: {
+  config: Config;
+  license: 'basic' | 'trial';
+  name: string;
+  kibanaConfig?: Record<string, any>;
+}) {
+  const services = config.get('services') as InheritedServices;
+  const servers = config.get('servers');
+  const kibanaServer = servers.kibana as UrlObject;
+  const apmSynthtraceKibanaClient = services.apmSynthtraceKibanaClient();
+  const allConfigs = config.getAll() as Record<string, any>;
+
+  const getScopedApiClientForUsername = (username: string) =>
+    getScopedApiClient(kibanaServer, username);
+
+  return {
+    ...allConfigs,
+    servers,
+    services: {
+      ...services,
+      getScopedApiClientForUsername: () => getScopedApiClientForUsername,
+      apmSynthtraceEsClient: (context: InheritedFtrProviderContext) =>
+        getApmSynthtraceEsClient(context, apmSynthtraceKibanaClient),
+    },
+    junit: {
+      reportName: `Observability AI Assistant API Integration tests (${name})`,
```

Comment:
It looks like we're creating an API integration test config here in the _functional_ test plugin. AFAIK, unlike UI tests, API integration tests aren't considered functional tests. Is this intended?